sklearn/pipeline.py

```diff
@@ -145,6 +145,25 @@ def fit(self, X, y=None, **fit_params):
         self.steps[-1][-1].fit(Xt, y, **fit_params)
         return self
 
+    def fit_predict(self, X, y=None, **fit_params):
```

Comment:
The `fit_predict` method should only be exposed if the underlying final estimator supports it as well. This can be implemented with the `@if_delegate_has_method` decorator as done later in that class on the `predict_proba` method:

```
@if_delegate_has_method(delegate='_final_estimator')
def fit_predict(self, X, y=None, **fit_params):
    ...
```